src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -2006,7 +2006,7 @@ RelationRestrictionForRelation(RangeTblEntry *rangeTableEntry,
 	List *filteredRelationRestrictionList =
 		filteredRelationRestrictionContext->relationRestrictionList;
 
-	if (list_length(filteredRelationRestrictionList) != 1)
+	if (list_length(filteredRelationRestrictionList) < 1)
```

Comment:
I guess this would work okay if the `relOptInfo` in all `relationRestriction` the same, if they contain different `relOptInfo` with either the query we are fixing or some other query then things might not work as expected, I guess we should verify that.

Still it is better than returning `NULL`

Reply:
We have looked at this with Ahmet and the expressions were the same in these relationRestrictions, the only difference was `varno` related things, so it seemed safe to choose any of these relationRestrictions, because we don't seem to be doing something with `varno` specific from these relationRestrictions.